src/v3/src/transformer/terminal/transformUnhandledErrors.ts

```diff
@@ -44,26 +51,41 @@ const getErrorMessage = (error?: AuthApiError, widgetProps?: WidgetProps) : stri
     },
     // special error messages
     {
-      tester: (err?: AuthApiError) => err?.errorCode === 'invalid_request' && err?.errorSummary === 'The recovery token is invalid',
+      tester: (err?: AuthApiError) => !!err?.errorCode && !!err?.errorSummary,
+      message: (err?: AuthApiError) => err!.errorSummary,
+    },
+  ];
+  const oauthErrorChecks: ErrorTester<OAuthError>[] = [
+    {
+      tester: (err?: OAuthError) => err?.error === 'invalid_request' && err?.error_description === 'The recovery token is invalid',
       message: () => loc('oie.invalid.recovery.token', 'login'),
     },
     {
-      tester: (err?: AuthApiError) => err?.errorCode === 'access_denied' && !!err?.errorSummary,
+      tester: (err?: OAuthError) => err?.error === 'access_denied' && !!err?.error_description,
       message: () => loc('oie.feature.disabled', 'login'),
     },
     {
-      tester: (err?: AuthApiError) => err?.errorCode && !!err?.errorSummary,
+      tester: (err?: OAuthError) => !!err?.error && !!err?.error_description,
       message: () => loc('oie.configuration.error', 'login'),
     },
-    // default fall back for unknown errors
-    {
-      tester: () => true,
-      message: () => loc('oform.error.unexpected', 'login'),
-    },
   ];
 
   // find the message that meets the tester condition
-  return errorChecks.find(({ tester }) => tester(error))?.message(error);
+  let message: string | undefined;
+  switch (error?.name) {
+    case 'AuthApiError':
+      message = authApiErrorChecks
+        .find(({ tester }) => tester(error as AuthApiError))?.message(error as AuthApiError);
+      break;
+    case 'OAuthError':
+      message = oauthErrorChecks
+        .find(({ tester }) => tester(error as OAuthError))?.message(error as OAuthError);
+      break;
+    default:
+      // intentionally fall through
+  }
+  // default fall back for unknown errors
+  return message || loc('oform.error.unexpected');
```

Comment:
nit: wondering if this is easier to follow?
```suggestion
    default:
      // default fall back for unknown errors
      message = loc('oform.error.unexpected');
  }
  return message;
```

Reply:
I considered this but the issue with this is if we enter one of the switch cases but the search for a message fails, then this `default` case doesn't get hit and we'll return `undefined`.

Reply:
Ahh, that makes complete sense. Ty!